ddtrace/writer.py

```diff
@@ -155,6 +168,19 @@ def _log_error_status(self, result, result_name):
                       getattr(result, "status", None), getattr(result, "reason", None),
                       getattr(result, "msg", None))
 
+    def _apply_processing_pipeline(self, traces):
+        if self._processing_pipeline is not None:
+            processed_traces = []
+            for trace in traces:
+                for processor in self._processing_pipeline:
+                    trace = processor.process_trace(trace)
+                    if trace is None:
+                        break
```

Comment:
Can we add a test for this case when the pipeline is short-circuited? You can have another test with two processors where the first one returns `None` so that the second is not invoked. If it helps, you can use Mock on the second one such as:
```python
processor = FilterRequestsOnUrl(r'http://example\.com/health')
processor = mock.Mock(processor, wraps=processor)
eq_(processor.process_trace.call_count, 0)
```